Spatial.gd

```diff
@@ -692,47 +692,56 @@ func gen_adjustment_nodes():
-		new_gizmo.connect("updated", icon, "update_point_poistion")
+		new_gizmo.connect("selected", self, "on_icon_gizmo_selected", [icon])
+		new_gizmo.connect("deselected", self, "on_gizmo_deselected")
+		new_gizmo.connect("updated", icon, "set_position")
 		$Gizmos.add_child(new_gizmo)
 
 var currently_selected_gizmo = null
-var currently_selected_node3d = null
-var currently_selected_node2d = null
+var currently_selected_icon = null
+var currently_selected_path3d = null
+var currently_selected_path2d = null
 var currently_selected_index = 0
 
-func on_gizmo_selected(object, node3d, point_type: String, node2d = null, index: int = 0):
+func on_icon_gizmo_selected(object: Spatial, icon: Sprite3D):
 	self.currently_selected_gizmo = object
-	self.currently_selected_node3d = node3d
-	self.currently_selected_node2d = node2d
+	self.currently_selected_icon = icon
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/DeleteNode.disabled = false
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/SnapSelectedToPlayer.disabled = false
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/XZSnapToPlayer.disabled = false
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/YSnapToPlayer.disabled = false
+
+
+func on_path_gizmo_selected(object: Spatial, path3d: Spatial, path2d: Line2D, index: int):
+	self.currently_selected_gizmo = object
+	self.currently_selected_path3d = path3d
+	self.currently_selected_path2d = path2d
 	self.currently_selected_index = index
 	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/DeleteNode.disabled = false
-	# Only enable these buttons if the object selected is a point on the path not an icon
-	if point_type == "path":
-		$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/NewNodeAfter.disabled = false
-		$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/ReversePathDirection.disabled = false
-		$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/SetActivePath.disabled = false
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/NewPathPointAfter.disabled = false
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/ReversePathDirection.disabled = false
+	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/SetActivePath.disabled = false
 	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/SnapSelectedToPlayer.disabled = false
 	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/XZSnapToPlayer.disabled = false
 	$Control/Dialogs/NodeEditorDialog/ScrollContainer/VBoxContainer/YSnapToPlayer.disabled = false
 
 
-func on_gizmo_deselected(object):
+func on_gizmo_deselected():
 	self.currently_selected_gizmo = null
-	self.currently_selected_node3d = null
-	self.currently_selected_node2d = null
+	self.currently_selected_icon = null
+	self.currently_selected_path3d = null
+	self.currently_selected_path2d = null
 	self.currently_selected_index = 0
```

Comment:
I think this should be null instead of 0 or any other number because there is no currently selected index. And even negative numbers are valid indexes